tests/test_custom_checkers/test_redundant_assignment_checker.py

```diff
@@ -244,3 +244,35 @@ def test_augassign_redundant(self):
             ignore_position=True,
         ):
             self.checker.visit_augassign(augassign_node)
+
+    def test_parallel_assign_redundant(self):
+        src = """
+        x, y = 0, 0
+        x, y = 10, 10
+        """
+        mod = astroid.parse(src)
+        mod.accept(CFGVisitor())
+        assign_node, *_ = mod.nodes_of_class(nodes.Assign)
+
+        self.checker.visit_module(mod)
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(msg_id="redundant-assignment", node=assign_node),
+            ignore_position=True,
+        ):
+            self.checker.visit_assign(assign_node)
+
+    def test_parallel_assign_one_variableredundant(self):
+        src = """
+        x, y = 0, 0
+        y = 10
+        """
+        mod = astroid.parse(src)
+        mod.accept(CFGVisitor())
+        assign_node, *_ = mod.nodes_of_class(nodes.Assign)
+
+        self.checker.visit_module(mod)
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(msg_id="redundant-assignment", node=assign_node),
```

Comment:
More importantly, however, this test reveals an issue with handling parallel assignment in this way. The error message (calling the assignment statement redundant) is misleading, since only the assignment to `y` is redundant, but not the assignment to `x`. This is a bit more challenging to fix, it might be a special case where only the `y` is highlighted (i.e., passing in the `AssignName` as the `node` attribute to the message, but only in the special case of parallel assignment where not every target is redundant.

Reply:
Hi Professor, I'd like to know how you expect the one reassigned variable in parallel assignment to be represented differently. Currently the rendering highlights the whole assignment statement, which could be misleading in this case. We could adjust the rendering to highlight just the` y` variable, but this might look inconsistent compared to how we handle other cases where the whole assignment is highlighted.

Another approach would be to leave the rendering as it is, but update the error message to specify which variables are redundant. For example, we could say, "The assignment of variable `y` is redundant". Please let me know which approach you would prefer, or if you have any other suggestions.

Reply:
@Raine-Yang-UofT Yes, these are great points. I'm fine with changing the error message to specify the variable(s) whose assignment is redundant, but please make sure this change applies to regular assignment statements as well. That is, you can make this modification to _all_ uses of this message, not just for parallel assignments.